stream.go

```diff
@@ -52,14 +131,34 @@ func (c *Conn) Write(r io.Reader) error {
 	return nil
 }
 
+// Recv get messages back from rev
+func (c *Conn) Recv() (*StreamMessage, error) {
+	select {
+	case err := <-c.err:
+		return nil, err
+	case msg := <-c.msg:
+		if msg.Type == "final" {
```

Comment:
Since c.msg is is unbuffred, this could be missed if recv is not being called when the final msg comes in.